mk/tools.mk

```diff
@@ -31,8 +31,15 @@ ifeq ($(OSFAMILY), linux)
 endif
 
 ifeq ($(OSFAMILY), macosx)
-  ARM_SDK_URL := https://developer.arm.com/-/media/Files/downloads/gnu/13.3.rel1/binrel/arm-gnu-toolchain-13.3.rel1-darwin-x86_64-arm-none-eabi.tar.xz
-  ARM_SDK_DIR := $(ARM_SDK_BASE_DIR)-darwin-x86_64-arm-none-eabi
+  # Check for Apple Silicon
+  UNAME_PROCESSOR := $(shell uname -p)
```

Comment:
IMO you can use `arch` (or `uname -m`) to get x86_64 / arm64 directly. And it may work on linux too, supporting aarch64 too. Not sure about windows, I have both uname and arch in path.

Reply:
https://superuser.com/a/1441469
```
ifeq ($(OS),Windows_NT)
    ifeq ($(PROCESSOR_ARCHITECTURE),AMD64)
```
need validation, the `if windows` is 9 years ago info.

Reply:
There is only 64bit i686 toolchain, so testing will only provide 'sorry' message instead of failing

Reply:
@ledvinap see #14073